Dynatrace/dynatrace-operator
# Start using deadcode `getDynakubes` from cmd/troubleshoot/builder.go

## Description

This PR started using deadcode `getDynakubes`, which by mistake was not used, it was refactored in this [PR]( https://github.com/Dynatrace/dynatrace-operator/pull/2164/files#diff-1da7a0324f5b6e77f575fd06f67abeb314f0114cba5718a85706fd36db6ca3adL122-R112) and introduced the bug of not using the proper dynakube name.


related ticket [K8S-10962](https://dt-rnd.atlassian.net/browse/K8S-10962)

## How can this be tested?

unit-tests and deploy few dynakubes and try to collect support_archive from only specific DK name.

After change:
```
bash-5.1$ dynatrace-operator troubleshoot --dynakube andrii
{"level":"info","ts":"2024-08-27T13:31:32.526Z","logger":"version","msg":"dynatrace-operator","version":"snapshot-remove-deadcode-get-dynakubes","gitCommit":"4e29f8f8f66ef243894a74810e1a8ed15ce9515d","buildDate":"2024-08-27T13:14:59+00:00","goVersion":"go1.23.0","platform":"linux/amd64"}
{"level":"info","ts":"2024-08-27T13:31:32.527Z","msg":"logging level","logLevel":"info"}
[oneAgentAPM]   --- checking if OneAgentAPM object exists ...
[oneAgentAPM]    ✓  OneAgentAPM does not exist
[namespace ]    --- checking if namespace 'dynatrace' exists ...
[namespace ]     ✓  using namespace 'dynatrace'
[          ]     X  error during getting dynakues: selected 'dynatrace:andrii' Dynakube does not exist: dynakubes.dynatrace.com "andrii" not found
```

before change (as you can see we ignore dkName) and just fetch any random DK from ns):
```
bash-5.1$ dynatrace-operator troubleshoot --dynakube andrii
{"level":"info","ts":"2024-08-27T15:01:18.125Z","logger":"version","msg":"dynatrace-operator","version":"snapshot","gitCommit":"d0a7ba28f2c47b07d4281fd9830209855f8eeaed","buildDate":"2024-08-27T13:26:01+00:00","goVersion":"go1.23.0","platform":"linux/amd64"}
{"level":"info","ts":"2024-08-27T15:01:18.126Z","msg":"logging level","logLevel":"info"}
[oneAgentAPM]   --- checking if OneAgentAPM object exists ...
[oneAgentAPM]    ✓  OneAgentAPM does not exist
[namespace ]    --- checking if namespace 'dynatrace' exists ...
[namespace ]     ✓  using namespace 'dynatrace'
[crd       ]    --- checking if CRD for Dynakube exists ...
[crd       ]     ✓  CRD for Dynakube exists
[dynakube  ]    --- checking if 'dynatrace:dynakube' Dynakube is configured correctly
[dynakube  ]        using 'dynatrace:dynakube' Dynakube
[dynakube  ]        secret token 'apiToken' exists
[dynakube  ]        checking if token scopes are valid
[dynakube  ]        token scopes are valid
[dynakube  ]        checking if can pull latest agent version
[dynakube  ]        API token is valid, can pull latest agent version
[dynakube  ]        pull secret 'dynatrace:azurecr' exists
[dynakube  ]        secret token '.dockerconfigjson' exists
[dynakube  ]     ✓  'dynatrace:dynakube' Dynakube is valid
```
